topics/reference_guide/custom_language_support/language_server_protocol.md

```diff
@@ -0,0 +1,183 @@
+Therefore, the LSP approach shouldn't be considered as a replacement for the existing language API, but rather as an added value.
+
+Starting with the 2023.2 release cycle, the LSP API is publicly available as part of the IntelliJ Platform in the following IDEs:
+- IntelliJ IDEA Ultimate
+- WebStorm
+- PhpStorm
+- PyCharm Professional
+- DataSpell
+- RubyMine
+- CLion
+- Aqua
+- DataGrip
+- GoLand
+- Rider
+
+## Plugin configuration
+
+To fully utilize the Language Server Protocol API in a third-party plugin based on the [](tools_gradle_intellij_plugin.md), we recommend upgrading the Gradle IntelliJ Plugin to the latest version available.
+This plugin will attach the LSP API sources and code documentation to your project.
+
+As LSP became available in the 2023.2 EAP7 of IntelliJ-based IDEs, the plugin must target IntelliJ IDEA Ultimate `2023.2` or later.
+
+Example <path>build.gradle.kts</path> configuration:
+
+```kotlin
+plugins {
+  // ...
+  id("org.jetbrains.intellij") version "1.15.0"
+}
+
+intellij {
+  version = "232-EAP-SNAPSHOT"
+  type = "IU"
+}
+```
+
+For projects based on the [](plugin_github_template.md), update the Gradle IntelliJ Plugin to the latest version, and amend the <path>gradle.properties</path> file as follows:
+
+```properties
+platformType = IU
+platformVersion = 232-EAP-SNAPSHOT
+pluginSinceBuild = 232
+pluginUntilBuild = 232.*
+```
+
+The <path>plugin.xml</path> configuration file needs to specify the dependency on the IntelliJ IDEA Ultimate module:
+
+```xml
+<idea-plugin>
+   <!-- ... -->
+   <depends>com.intellij.modules.platform</depends>
+   <depends>com.intellij.modules.ultimate</depends>
+</idea-plugin>
+```
+
+The LSP API sources are bundled in IntelliJ IDEA Ultimate and can be found within the <path>[IDE]/lib/src/src_lsp-openapi.zip</path> archive.
+
+## Supported features
+
+The initial LSP support within the IntelliJ Platform covers the following features:
+
+- Errors/warnings highlighting ([textDocument/publishDiagnostics](https://microsoft.github.io/language-server-protocol/specification/#textDocument_publishDiagnostics))
+- Quick-fixes for these errors/warnings ([textDocument/codeAction](https://microsoft.github.io/language-server-protocol/specification/#textDocument_codeAction))
+- Code completion ([textDocument/completion](https://microsoft.github.io/language-server-protocol/specification/#textDocument_completion))
+- Go to Declaration ([textDocument/definition](https://microsoft.github.io/language-server-protocol/specification/#textDocument_definition))
+- More features will be announced soon.
+
+## Basic implementation
+
+A minimal LSP integration must implement `LspServerSupportProvider` along with a service descriptor and register it as a `com.intellij.platform.lsp.serverSupportProvider` extension point:
```

Comment:
"service descriptor" -> `LspServerDescriptor`?
Please also add links to these classes and make sure other class names have links on first mention.